### PR TITLE
fix for #80: call unittest from setuptools

### PR DESCRIPTION
**Pull request by WeinaJi**

### CODE REVIEW

File: ci/python_test.sh [hunk 1]
```diff
@@ -12,10 +12,16 @@ fi
...
-$BIN/pip install nose
+pip -v install --force .
+pip install nose
 
-$BIN/nosetests -s -v python/tests
+nosetests -s -v -P python/tests
+
+PYTHON_MAJOR_VERSION=$(python -c 'import sys; print(sys.version_info[0])')
+if [[ $PYTHON_MAJOR_VERSION -ge 3 ]]; then
+    python setup.py test
```
Comment (WeinaJi) [inline ci/python_test.sh:26]: The python test would trigger both unittest and doc test. I don't know where in the package it sets sphinx > 2.0, the doc build always failed for python 2. So I skip the test for python 2.

Comment (matz-e) [reply ci/python_test.sh:26]: Seems reasonable to me. IMO, Python 2 support should be mostly best effort by now, anyways.

File: setup.py [hunk 2]
```diff
@@ -173,7 +176,8 @@ def run(self):
         test_doc=get_sphinx_command,
     ),
     zip_safe=False,
-    setup_requires=setup_requires,
+    setup_requires=setup_requires + doc_requires if "test" in sys.argv or "test_doc" in sys.argv
+    else setup_requires,
```
Comment (matz-e) [inline setup.py:180]: This seems a little odd formatting. What would `black` do here?

**Review by matz-e — COMMENTED** [review]
Looks great to me, thank you!

I'll give @mgeplf a chance to comment, otherwise I'll approve in a few days.

**Review by matz-e — APPROVED** [review]

**Comment by WeinaJi** [comment]
Hello @matz-e @mgeplf ,  can I merge this PR ?

**Comment by matz-e** [comment]
Yes from my side. It's been open long enough

File: python/tests/__init__.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import unittest
+
+
+def test_suite():
+    loader = unittest.TestLoader()
+    suite = loader.discover('python/tests', pattern='test*.py')
```
Comment (mgeplf) [inline python/tests/__init__.py:6]: FYI, this hard-coded path meant the tests can only be run from the root of the repository: I fixed it with:
https://github.com/BlueBrain/libsonata/pull/106

**Pull request merged**